src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -68,6 +68,29 @@ class IpexCommunicationService extends AgentService {
   @OnlineOnly
   async acceptAcdc(id: string): Promise<void> {
     const grantNoteRecord = await this.getNotificationRecordById(id);
+    const notificationRecord = await this.notificationStorage.findById(
+      grantNoteRecord.id
+    );
+
+    if (!notificationRecord) {
+      throw new Error(
+        `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+      );
+    }
+
+    if (
+      notificationRecord?.id &&
```

Comment:
Do we need this line, won't it always be defined?

Reply:
now we have `grantNoteRecord &&` - we don't need this line, we just checked `if (!grantNoteRecord)` above it